packages/@react-aria/selection/src/useHighlightSelectionDescription.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2021 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {AriaLabelingProps} from '@react-types/shared';
+// @ts-ignore
+import intlMessages from '../intl/*.json';
+import {MultipleSelectionManager} from '@react-stately/selection';
+import {useDescription} from '@react-aria/utils';
+import {useInteractionModality} from '@react-aria/interactions';
+import {useMemo} from 'react';
+import {useMessageFormatter} from '@react-aria/i18n';
+
+interface UseHighlightSelectionDescriptionProps {
+  selectionManager: MultipleSelectionManager,
+  hasItemActions?: boolean
+}
+
+/**
+ * Computes the description for a grid selectable collection.
+ * @param props
+ */
+export function useHighlightSelectionDescription(props: UseHighlightSelectionDescriptionProps): AriaLabelingProps {
```

Comment:
Sorry, but can we move this into the `grid` package for now since it's the only place it's used? I just don't want to be locked into this API and have the option to move it later.